src/statistics.jl

```diff
@@ -0,0 +1,46 @@
+abstract type AbstractStatistic end
+
+struct Mean{D<:AbstractDomain} <: AbstractStatistic
+   domain::D
+   values::Dict{Symbol, Vector} 
+end
+
+Mean(domain, values) = Mean{typeof(domain)}(domain, values)
+
+function Mean(solution::SimulationSolution)
+    # solution variables
+    variables = collect(keys(solution.realizations))
+
+    # output dictionary
+    values = Dict(var => mean(solution.realizations[var]) for var in variables)
+    
+    Mean(solution.domain, values)
+end
+
+mean(solution::SimulationSolution) = Mean(solution)
+
+function digest(statistic::Mean{<:RegularGrid})
```

Comment:
Let's think about `digest` later, I think we could eliminate it from this PR for now.